C/decoder/decode.c

```diff
@@ -0,0 +1,200 @@
+#include "decode.h"
+
+#define NUMERIC_ASCII_START 48
+#define UPPER_ALPHA_ASCII_START 65
+#define LOWER_ALPHA_ASCII_START 97
+
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+static char chars[83] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz#$%*+,-.:;=?@[]^_{|}~";
+
+/*
+	Image processing functions start
+*/
+
+static float sRGBToLinear(int value) {
+	float v = (float)value / 255;
+	if(v <= 0.04045) return v / 12.92;
+	else return powf((v + 0.055) / 1.055, 2.4);
+}
+
+static float signPow(float value, float exp) {
```

Comment:
stick to either camelCase or snake_case

Reply:
Made all camelCase to maintain uniformity.